drivers/sensor/iis328dq/iis328dq.c

```diff
@@ -0,0 +1,460 @@
+	case IIS328DQ_ATTR_DURATION:
+		if (chan != SENSOR_CHAN_ACCEL_XYZ) {
+			LOG_ERR("Duration cannot be set per-channel");
+			return -ENOTSUP;
+		}
+		return iis328dq_set_duration(dev, val->val1);
+#endif /* CONFIG_IIS328DQ_THRESHOLD */
+	default:
+		LOG_DBG("Acc attribute not supported");
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis328dq_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+	case SENSOR_CHAN_ALL:
+		return iis328dq_dev_config(dev, chan, attr, val);
+	default:
+		LOG_DBG("Attr not supported on %d channel", chan);
+		break;
+	}
+
+	return -ENOTSUP;
+}
+
+static int iis328dq_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct iis328dq_data *iis328dq = dev->data;
+	const struct iis328dq_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	int16_t buf[3];
+
+	/* fetch raw data sample */
+	if (iis328dq_acceleration_raw_get(ctx, buf) < 0) {
+		LOG_DBG("Failed to fetch raw data sample");
+		return -EIO;
+	}
+
+	iis328dq->acc[0] = buf[0] >> 4;
+	iis328dq->acc[1] = buf[1] >> 4;
+	iis328dq->acc[2] = buf[2] >> 4;
+
+	return 0;
+}
+
+static const struct sensor_driver_api iis328dq_driver_api = {
+	.attr_set = iis328dq_attr_set,
+#if CONFIG_IIS328DQ_TRIGGER
+	.trigger_set = iis328dq_trigger_set,
+#endif /* CONFIG_IIS328DQ_TRIGGER */
+	.sample_fetch = iis328dq_sample_fetch,
+	.channel_get = iis328dq_channel_get,
+};
+
+static int iis328dq_init(const struct device *dev)
+{
+	struct iis328dq_data *iis328dq = dev->data;
+	const struct iis328dq_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	uint8_t wai;
+
+	iis328dq->dev = dev;
+
+	/* check chip ID */
+	if (iis328dq_device_id_get(ctx, &wai) < 0) {
+		return -EIO;
+	}
+
+	if (wai != IIS328DQ_ID) {
+		LOG_ERR("Invalid chip ID");
+		return -EINVAL;
+	}
+
+	/* reset device */
+	if (iis328dq_boot_set(ctx, PROPERTY_ENABLE) < 0) {
+		return -EIO;
+	}
+
+	k_sleep(K_MSEC(100));
+
+	if (iis328dq_boot_get(ctx, &wai) < 0) {
+		return -EIO;
+	}
+	if (wai != PROPERTY_DISABLE) {
+		LOG_ERR("BOOT did not deassert");
+		return -EIO;
+	}
+
+	if (iis328dq_block_data_update_set(ctx, PROPERTY_ENABLE) < 0) {
+		return -EIO;
+	}
+
+	/* set default odr to 12.5Hz acc */
+	if (iis328dq_set_odr(dev, 12) < 0) {
+		LOG_ERR("odr init error (12.5 Hz)");
+		return -EIO;
+	}
+
+	if (iis328dq_set_range(dev, cfg->range) < 0) {
+		LOG_ERR("range init error %d", cfg->range);
+		return -EIO;
+	}
+
+#ifdef CONFIG_IIS328DQ_TRIGGER
+	if (iis328dq_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupts");
+		return -EIO;
+	}
+#endif /* CONFIG_IIS328DQ_TRIGGER */
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "IIS328DQ driver enabled without any devices"
+#endif
```

Comment:
I don't think this is possible, isn't it?
In Kconfig you checked:

```
select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_IIS328DQ),i2c)
select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_IIS328DQ),spi)
```

So, there shuold be instances on either one of the bus. Correct?

Reply:
I suppose it's superfluous due to `depends on DT_HAS_ST_IIS328DQ_ENABLED`, I just copied it from `iis2dlpc`. I've removed it.